pulsar/producer_partition.go

```diff
@@ -863,13 +857,19 @@ func (p *partitionProducer) internalFlushCurrentBatch() {
 		return
 	}
 
+	p.writeData(batchData, sequenceID, &callbacks)
+}
+
+func (p *partitionProducer) writeData(buffer internal.Buffer, sequenceID uint64, callbacks *[]interface{}) {
```

Comment:
I think `[]interface{}` is better than `*[]interface{}`, in go, although the parameter is passed by value, but if the parameter is  a slice, it just copy the value of the address of the underlying array, not the value of all the elements in that array.

Reply:
Fixed